### PR TITLE
Update as_array returned shape and update detection pipeline test

### PR DESCRIPTION
**Pull request by Kh4L**
Signed-off-by: Serge Panev <spanev@nvidia.com>

### CODE REVIEW

**Comment by Kh4L** [comment]
Build 691133

**Comment by Kh4L** [comment]
Build 691239

**Comment by Kh4L** [comment]
Build 692189

**Comment by Kh4L** [comment]
Build 692299

File: dali/python/backend_impl.cc [hunk 1]
```diff
@@ -273,35 +273,32 @@ void ExposeTensorList(py::module &m) { // NOLINT
...
 
-          DALI_ENFORCE(IsValidType(t.type()), "Cannot produce "
-              "buffer info for tensor w/ invalid type.");
-          DALI_ENFORCE(t.IsDenseTensor(),
-                      "Tensors in the list must have the same shape");
+          if (IsValidType(t.type())) {
+            format = FormatStrFromType(t.type());
+            type_size = t.type().size();
+          } else {
+            // Default is float
```
Comment (klecki) [inline dali/python/backend_impl.cc]: Why? Is this something numpy does as well?

File: dali/test/python/test_detection_pipeline.py [hunk 2]
```diff
@@ -279,15 +279,7 @@ def set_iters(args, dataset_size):
...
-        # Temporary workaround for empty TensorListGPU containing boxes
-        # Will be addressed by DALI-646
-        try:
-            dali_out = dali_out.asCPU()
-        except:
-            return np.empty((0, 4), np.float32)
-
-        # Proper version
-        # dali_out = dali_out.asCPU()
+        dali_out = dali_out.asCPU()
```
Comment (jantonguirao) [inline dali/test/python/test_detection_pipeline.py]: I believe `as_cpu() ` is the proper syntax now. `asCPU` is deprecated

File: dali/python/backend_impl.cc [hunk 3]
```diff
@@ -273,35 +273,32 @@ void ExposeTensorList(py::module &m) { // NOLINT
...
-            // Based on default numpy empty array shape and strides
-            std::vector<ssize_t> shape(1, 0);
-            std::vector<ssize_t> strides(1, type_size);
-            return py::array(py::buffer_info(nullptr,
-                                             type_size,
-                                             std::move(format),
-                                             shape.size(),
-                                             shape,
-                                             strides));
+            raw_mutable_data = nullptr;
```
Comment (jantonguirao) [inline dali/python/backend_impl.cc]: initialize `raw_mutable_data = nullptr` in line 276 and here just do `if (t.size() > 0)` for what is now the `else`

File: dali/test/python/test_detection_pipeline.py (hunk 2 above)
Comment (Kh4L) [reply dali/test/python/test_detection_pipeline.py]: It is

File: dali/python/backend_impl.cc (hunk 3 above)
Comment (Kh4L) [reply dali/python/backend_impl.cc]: Done

File: dali/test/python/test_detection_pipeline.py (hunk 2 above)
Comment (Kh4L) [reply dali/test/python/test_detection_pipeline.py]: Done

File: dali/python/backend_impl.cc (hunk 1 above)
Comment (Kh4L) [reply dali/python/backend_impl.cc]: Yes

**Comment by Kh4L** [comment]
Build 692357

**Pull request merged**